src/openrct2/scripting/bindings/entity/ScGuest.cpp

```diff
@@ -481,8 +481,6 @@ namespace OpenRCT2::Scripting
             {
                 if (thought.type == PeepThoughtType::None)
                     break;
-                if (thought.freshness == 0)
```

Comment:
why remove this?

Reply:
I had done this as I figured that there was no need to check the thought freshness, but other code that iterates over thoughts has this condition so it makes more sense to not remove it. Regardless, this would have belonged in a different PR, but I'll just revert it as it makes more sense to leave this condition in.

Reply:
One thing I noticed from this is that the guest.thought event receives fresh thoughts which may not show up in the guest's thought list until at least the next game tick, or longer if another thought was very recently made fresh, according to https://github.com/OpenRCT2/OpenRCT2/blob/36b6e9a011c3423a793a40f63e2190de68fd8533/src/openrct2/entity/Peep.cpp#L906-L909

Maybe there could be another event for when a thought actually enters the thought list, or maybe that would be excessive.